### DISCUSSION TITLE
Debugging Xyce with Visual Studio Code

### DISCUSSION THREAD

**Discussion by Codester96** [Xyce/Xyce#57] [Q&A] [answered]
Has anyone had any experience in debugging Xyce using Visual studio code or any other programs?

It would be great if someone could tell me the correct configurations for launch.json, settings.json or tasks.json in VS Code.

Otherwise, I am also happy to work with other programs as long as debugging works.

Cheers,
John

**Comment by tvrusso** [comment, marked as answer]
Nobody on the Xyce team has used Visual Studio to debug Xyce.  It is rare that we ever do development or debugging on Windows directly.  There may be other people in the open source community who have used Visual Studio, and the place to ask about that would be the Xyce Google Group.

Most of our debugging is done on Linux and OS X, mostly using gdb as the debugger.

**Reply by Codester96** [reply]
Hi, thank you for your response.
I tried to debug Xyce on Ubuntu just now using Terminals.
However, I get an error message saying that "#include Xyce_config.h: No such file or directory" even though the program can be ran.
I typed "gcc -g -o xyce Xyce.C" on terminals, inside the src file from Xyce-Release.

Could you tell me what the problem is?

**Reply by tvrusso** [reply]
That's not how you do it.   Xyce is an enormous code and you cannot just try to compile a single file and expect it to work.  You build the code normally just as per the building guide, but use "-O0" and "-fno-inline" in the CXXFLAGS instead of "-O3".

After that you can run the finished build through gdb.

We cannot give you detailed instructions on how to use gcc and gdb, you'll have to find references and study up on them.
